main.go

```diff
@@ -138,26 +205,26 @@ func main() {
 
 	ctx := ctrl.SetupSignalHandler()
 
-	tlsOptions, err := common.GetSspTlsOptions(ctx)
-	if err != nil {
-		setupLog.Error(err, "Error while getting tls profile")
-		os.Exit(1)
-	}
+	var mgr ctrl.Manager
 
-	err = runPrometheusServer(metricsAddr, *tlsOptions)
-	if err != nil {
-		setupLog.Error(err, "unable to start prometheus server")
-		os.Exit(1)
+	getTLSOptsFunc := func(cfg *tls.Config) {
+		cfg.GetConfigForClient = func(_ *tls.ClientHelloInfo) (*tls.Config, error) {
+			// This is a hack, because webhook.Server does not implement InjectClient, so
```

Comment:
Please avoid this

Reply:
What do you recommend instead?
This is the easiest solution.

Reply:
The term hack drove me away initially, but after looking at this again this seems to be an elegant solution to me, so let's do it like this.

Reply:
Ok, @opokornyy can you remove this comment, or rewrite it?